tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -461,20 +475,52 @@ def _assert_pantsd_keyboardinterrupt_signal(self, signum, messages, regexps=[],
   def test_pantsd_sigterm(self):
     self._assert_pantsd_keyboardinterrupt_signal(
       signal.SIGTERM,
-      ['Signal {signum} (SIGTERM) was raised. Exiting with failure.'.format(signum=signal.SIGTERM)])
-
-  def test_keyboardinterrupt_signals_with_pantsd(self):
-    for interrupt_signal in [signal.SIGINT, signal.SIGQUIT]:
-      self._assert_pantsd_keyboardinterrupt_signal(
-        interrupt_signal,
-        ['\nInterrupted by user over pailgun client!\n'])
+      messages=[
+        "\nSignal {signum} (SIGTERM) was raised. Exiting with failure.\n"
+        .format(signum=signal.SIGTERM),
+      ],
+      regexps=["""
+Interrupted by user:
+Interrupted by user over pailgun client!
+$"""],
+      quit_timeout=5.0)
+
+  def test_pantsd_sigquit(self):
+    self._assert_pantsd_keyboardinterrupt_signal(
+      signal.SIGQUIT,
+      messages=[
+        "\nSignal {signum} (SIGQUIT) was raised. Exiting with failure.\n"
+        .format(signum=signal.SIGQUIT),
+      ],
+      regexps=["""
+Interrupted by user:
+Interrupted by user over pailgun client!
+$"""],
+      quit_timeout=5.0)
```

Comment:
Should we just bump the default value to 5.0, rather than overriding this in tests? Ideally the default value would cover the common cases we care about, and "on Travis" feels like a common case we care about...

Reply:
100% agree with this!

Reply:
Or actually, I agree with the part about bumping the default value, but I would love the default for interactive usage to be less than 5 seconds, since in the case where the pantsd-runner process is unresponsive except to a SIGKILL that leads to 5 seconds of the user not having control of their terminal. In CI in general however (not just in upstream pants) I think we absolutely want to increase this (likely to 5 seconds) since we don't have interactive users sending <kbd>ctrl-c</kbd> to pants.

I would love to set the default depending on whether stdin is a tty, I'll open up a separate issue to discuss that. For now I think the best of both worlds might be to increase the hardcoded default to 5 seconds, but also to immediately print an info log saying "pants is waiting <x> seconds for the subprocess to time out" before waiting for the timeout.

Reply:
Made #7514 for the possibility of decreasing the hardcoded default if at a tty. The point about:
> Ideally the default value would cover the common cases we care about, and "on Travis" feels like a common case we care about...

is great, and I'll try to remember that in the future. Thanks!